app/spell_check.py

```diff
@@ -20,6 +32,24 @@ def check_word(word_list):
 
     return results
 
+def check_other_lang(word_list, dictionary):
+    results = []
+    for idx, word in enumerate(word_list):
+        lower_word = str(unicodedata.normalize('NFC', word)).lower()
+        lower_word = lower_word.strip()
+        if lower_word not in dictionary and lower_word not in ',.!@#$/;:\'':
+            results.append(idx)
+    return results
+
 def word_candidates(word_to_check):
     check = SpellChecker()
-    return check.candidates(word_to_check)
\ No newline at end of file
+    return check.candidates(word_to_check)
+
+def recombine(word_list):
```

Comment:
What is this function used for? I see it being called at the end of index_page() but I'm not following why. This recombining only a few specific symbols with the previous word? Is this something unique in Irish?

Reply:
This is mostly the way I was dealing with whitespaces popping up between words ad following special characters such as : ? , etc. Given the special parsing I had to do for this the special characters were appearing as individual elements in the results_words list and when printed were appearing with spaces before them

Reply:
Okay I think I'm following now. However, once this is updated to support the refactored tokenization like we are discussing in the following comment, I think that this would become unnecessary. What do you think?

Reply:
I imagine it would, once we get the updated tokenization in i can remove that

Reply:
removed/replaced in other PR